### ISSUE TITLE
Impersonated kirby user permissions are wrong after page update within loop

### ISSUE THREAD

**Issue by LeBenLeBen** [getkirby/kirby#2830]
**Describe the bug**  
I have a script that impersonate the user with `kirby` and loops over a collection of pages to update them. The first page update works but after that, all others fail with the error:

> You are not allowed to update "page-slug"

Running the script again works for the first page (which previously failed) and fail again for the next pages.

This used to work in the past and can be reproduced at least since Kirby 3.4.0.

**To Reproduce**  
The script looks like this:

```php
$kirby->impersonate('kirby');

$pages = $kirby->page('blog')->index();

foreach ($kirby->languages() as $language) {
  foreach ($pages as $page) {
    try {
      $page->update(
        [
          'expertises' => ['foo', 'bar'],
        ],
        $language->code()
      );
    } catch (Exception $e) {
      echo $e->getMessage() . "\n";
    }
  }
}
```

**Expected behavior**  
The update of all the pages in the collection should pass.

**Kirby Version**  
3.4.3

**Additional context**  
There seems to be something that reset the objects permissions after the first page update.

Before first page update it looks like this:

```
object(Kirby\Cms\PagePermissions)#16984 (11) {
  ["changeSlug"]=>  bool(true)
  ["changeStatus"]=>  bool(true)
  ["changeTemplate"]=>  bool(false)
  ["changeTitle"]=>  bool(true)
  ["create"]=>  bool(true)
  ["delete"]=>  bool(true)
  ["duplicate"]=>  bool(true)
  ["read"]=>  bool(true)
  ["preview"]=>  bool(true)
  ["sort"]=>  bool(false)
  ["update"]=>  bool(true)
}

object(Kirby\Cms\UserPermissions)#16984 (8) {
  ["create"]=>  bool(false)
  ["changeEmail"]=>  bool(true)
  ["changeLanguage"]=>  bool(true)
  ["changeName"]=>  bool(true)
  ["changePassword"]=>  bool(true)
  ["changeRole"]=>  bool(true)
  ["delete"]=>  bool(true)
  ["update"]=>  bool(true)
}
```

But then it becomes:

```
object(Kirby\Cms\PagePermissions)#26477 (11) {
  ["changeSlug"]=>  bool(false)
  ["changeStatus"]=>  bool(false)
  ["changeTemplate"]=>  bool(false)
  ["changeTitle"]=>  bool(false)
  ["create"]=>  bool(false)
  ["delete"]=>  bool(false)
  ["duplicate"]=>  bool(false)
  ["read"]=>  bool(false)
  ["preview"]=>  bool(false)
  ["sort"]=>  bool(false)
  ["update"]=>  bool(false)
}

object(Kirby\Cms\UserPermissions)#26409 (8) {
  ["create"]=>  bool(false)
  ["changeEmail"]=>  bool(false)
  ["changeLanguage"]=>  bool(false)
  ["changeName"]=>  bool(false)
  ["changePassword"]=>  bool(false)
  ["changeRole"]=>  bool(false)
  ["delete"]=>  bool(false)
  ["update"]=>  bool(false)
}
```

**Comment by lukasbestle** [comment]
I have tested this with the current `features` branch, but unfortunately I cannot reproduce this.

My example code:

```php
$kirby->impersonate('kirby');

foreach ($site->index() as $p) {
    $p->update(['test' => uniqid()]);
}
```

I've also tried a multilang setup:

```php
$kirby->impersonate('kirby');

$index = $site->index();

foreach ($kirby->languages() as $language) {
    foreach ($index as $p) {
        $p->update(['test' => uniqid()], $language->code());
    }
}
```

Both work fine.

@LeBenLeBen Can you reproduce the issue in the Starterkit? Maybe something special in your project is causing this.

**Comment by LeBenLeBen** [comment]
According to my tests, the regression has been introduced in Kirby 3.3.2. So I looked at the changes made in this version and came across #2323

I removed the `num` parameter from my `article.yml` blueprint and the script worked again.

Could this fix for `delete` introduced this bug for `update`? Or was it introduced prior and fixed only for `delete`?

BTW my custom `num` parameter looks like this: `num: '{{ page.date.toDate("%G%m%d") }}'` (yes, I use `'date.handler' => 'strftime'`)

**Comment by lukasbestle** [comment]
Ah, then it's most likely the same issue that we will fix in Kirby 3.5.0.

Could you please check if https://github.com/getkirby/kirby/pull/2894 fixes it for you?

**Comment by LeBenLeBen** [comment]
@lukasbestle Yes it does! 🙌

**Comment by lukasbestle** [comment]
Thanks for testing! I'm closing this issue as it will be fixed in 3.5.0.
